packages/@aws-cdk/aws-servicediscovery/lib/instance.ts

```diff
@@ -0,0 +1,47 @@
+import cdk = require('@aws-cdk/cdk');
+import { IService } from './service';
+
+export interface IInstance extends cdk.IConstruct {
+  /**
+   * The id of the instance resource
+   */
+  readonly instanceId: string;
+
+  /**
+   * The Cloudmap service this resource is registered to.
+   */
+  readonly service: IService;
+}
+
+/**
+ * Used when the resource that's associated with the service instance is accessible using values other than an IP
+ * address or a domain name (CNAME), i.e. for non-ip-instances
+ */
+export interface BaseInstanceProps {
+  /**
+   * The id of the instance resource
+   *
+   * @default CloudFormation-generated name
```

Comment:
In fact the default is a CDK-generated name in this case, which is guaranteed to be unique for a given location in the construct tree.